### PR TITLE
feat(LINGUIST-358): Onboarding screen for chat

### PR DESCRIPTION
**Pull request by selimcanglr**
It will only appear if there are 0 messages and it is the first time the user is entering the screen.

### CODE REVIEW

File: screens/chat/ChatScreen.tsx [hunk 1]
```diff
@@ -183,15 +210,24 @@ const ChatScreen = ({ route }: ChatScreenProps) => {
...
-              <ChatMessageComponent
-                onWordPress={handleWordPress}
-                key={item.id || item.timestamp.toString()}
-                chatMessage={item}
-              />
-            ))}
+          <CopilotStep
+            name="chat-message-list"
+            order={6}
+            text="Your messages will apear here. You can click on a word to see the word's definitions."
```
Comment (can-aslan) [inline screens/chat/ChatScreen.tsx:216]: "apear" -> "ap`p`ear"

"see the word's definitions" -> "see the word's definition(s)"

File: screens/chat/ChatScreen.tsx [hunk 2]
```diff
@@ -210,18 +246,30 @@ const ChatScreen = ({ route }: ChatScreenProps) => {
...
       </Modal>
-      <View style={styles.header}>
-        <ChatHeader />
-      </View>
+      <CopilotStep
+        name="chat-screen"
+        order={1}
+        text={`This is the chat screen, you are talking with "${
+          currentBot?.name || 'a chatbot'
+        }". You can ask questions or just chat with it.`}
```
Comment (can-aslan) [inline screens/chat/ChatScreen.tsx:254]: "or just chat with it" -> "or just chat about anything" OR "or just chat" OR some variation you deem fit

**Review by can-aslan — CHANGES_REQUESTED** [review]
Left some UX and typo comments

**Comment by can-aslan** [comment]
Also `npm install` fails on this branch, and only `npm install --force` works. Do you also have this issue, or is this just for me? Can you check?

**Comment by selimcanglr** [comment]
> Also `npm install` fails on this branch, and only `npm install --force` works. Do you also have this issue, or is this just for me? Can you check?

Try npx expo install

**Comment by KardelenCeren** [comment]
How can I test this? 
I tried `npm install`, `npx expo install` and then `npx expo install --fix` since it prompted me, but no luck. I used a newly registered account and entered the chat screen on this branch.

File: components/chat/ChatMenu.tsx [hunk 3]
```diff
@@ -43,18 +48,29 @@ const ChatMenu = ({ menuVisible, setMenuVisible, triggerOption }: ChatMenuProps)
...
-            <View style={styles.optionContent}>
-              {option.icon}
-              <Text style={styles.option}>{option.label}</Text>
-            </View>
-          </MenuOption>
-        ))}
+        <CopilotStep
+          name="chat-menu-options"
+          order={4}
+          text="You can click on 'See Active Words' to see the active words used in this conversation.  Active words will be used by the chatbot more frequently so that you can learn bout them. The words are selected from your wordlists on a daily basis."
```
[the remark this answers is not in the document]
Comment (selimcanglr) [reply components/chat/ChatMenu.tsx:54]: Changed it

File: components/chat/ChatHeader.tsx [hunk 4]
```diff
@@ -65,15 +75,25 @@ const ChatHeader = () => {
...
-            setMenuVisible={setChatMenuVisible}
-            triggerOption={triggerOption}
-          />
-          {activeWordsVisible && (
-            <ActiveWordsModal setVisible={setActiveWordsVisible} visible={activeWordsVisible} />
-          )}
+          <CopilotStep
+            name="chat-quiz-button"
+            order={2}
+            text="The 'Start Quiz' button will be enabled after you chat for a certain amount with the bot. It will send you to a quiz to test your knowledge of the words used in the conversation."
```
[the remark this answers is not in the document]
Comment (selimcanglr) [reply components/chat/ChatHeader.tsx:81]: Changed it

File: screens/chat/ChatScreen.tsx (hunk 1 above)
Comment (selimcanglr) [reply screens/chat/ChatScreen.tsx:216]: Fixed

**Comment by KardelenCeren** [comment]
Yess now it works and it looks great!! Maybe some texts can be shorter? Youtube shorts significantly decreased my attention span :/
Also on my device the highlighted regions are not where they are supposed to be: 
image

**Comment by selimcanglr** [comment]
Yes highlighting is a bit off because of either the library or my incorrect usage. Let's send it this way for now.

**Pull request merged**